samza-core/src/main/java/org/apache/samza/storage/StoreBackupEngine.java

```diff
@@ -0,0 +1,185 @@
+package org.apache.samza.storage;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import org.apache.samza.SamzaException;
+import org.apache.samza.config.Config;
+import org.apache.samza.metrics.MetricsRegistryMap;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.apache.hadoop.fs.FileStatus;
+import org.apache.hadoop.fs.FileSystem;
+import org.apache.hadoop.fs.Path;
+import org.apache.hadoop.conf.*;
+
+// Class to hold common methods and variables for backup storage and restore process
+public class StoreBackupEngine {
+  protected static final Logger LOGGER = LoggerFactory.getLogger(StoreBackupEngine.class);
+  protected FileSystem fs;
+  protected Path hdfsPath;
+  protected Path dbPath;
+  public StoreBackupMetrics metrics;
+
+  public StoreBackupEngine(Config storeConfig) {
+    metrics = new StoreBackupMetrics("backup", new MetricsRegistryMap("registrymap"));
+
+    // Setup fs configuration settings
+    Configuration conf = new Configuration();
+    // If core site path file is specified
+    if (storeConfig.containsKey("hdfs.coreSitePath")){
+      Path coreSitePath = new Path(storeConfig.get("hdfs.coreSitePath"));
+      conf.addResource(coreSitePath);
```

Comment:
Do you need the entire properties file or only a couple of configs? If the latter, let's add them to configs separately instead of requiring the entire file.